deeplake/core/vectorstore/deeplake_vectorstore.py

```diff
@@ -0,0 +1,213 @@
+        if ids:
+            view = self.dataset.filter(lambda x: x["ids"].data()["value"] in ids)
+            ids = list(view.sample_indices)
+
+        if filter:
+            if view is None:
+                view = self.dataset
+            view = view.filter(partial(utils.dp_filter, filter=filter))
+            ids = list(view.sample_indices)
+
+        with self.dataset:
+            for id in sorted(ids)[::-1]:
+                self.ds.pop(id)
+            self.dataset.commit(f"deleted {len(ids)} samples", allow_empty=True)
+        return True
+
+    @classmethod
+    def force_delete_by_path(cls, path: str) -> None:
+        """Force delete dataset by path"""
+        try:
+            import deeplake
+        except ImportError:
+            raise ValueError(
+                "Could not import deeplake python package. "
+                "Please install it with `pip install deeplake`."
+            )
+        deeplake.delete(path, large_ok=True, force=True)
+
+    def _exact_text_search(self, view, query):
+        view = view.filter(lambda x: query in x["text"].data()["value"])
+        scores = [1.0] * len(view)
+        index = view.index.values[0].value[0]
+        return (view, scores, index)
+
+    def _attribute_based_filtering(self, filter):
+        view = self.dataset
+        # attribute based filtering
+        if filter is not None:
+            if isinstance(filter, dict):
+                filter = partial(utils.dp_filter, filter=filter)
+
+            view = view.filter(filter)
+            if len(view) == 0:
+                return []
+        return view
+
+    def _run_data_injestion(self, elements):
+        batch_size = min(self.ingestion_batch_size, len(elements))
+        if batch_size == 0:
+            return []
+
+        batched = [
+            elements[i : i + batch_size] for i in range(0, len(elements), batch_size)
+        ]
+
+        self.ingest().eval(
+            batched,
+            self.dataset,
+            num_workers=min(self.num_workers, len(batched) // max(self.num_workers, 1)),
+            _embedding_function=self._embedding_function,
+        )
+
+    @deeplake.compute
+    def ingest(sample_in: list, sample_out: list, _embedding_function) -> None:
+        text_list = [s["text"] for s in sample_in]
+
+        embeds = [None] * len(text_list)
+        if _embedding_function is not None:
+            embeddings = _embedding_function.embed_documents(text_list)
+            embeds = [np.array(e, dtype=np.float32) for e in embeddings]
+
+        for s, e in zip(sample_in, embeds):
+            embedding = e if _embedding_function else sample_in["embedding"]
+            sample_out.append(
+                {
+                    "text": s["text"],
+                    "metadata": s["metadata"],
+                    "ids": s["ids"],
+                    "embedding": embedding,
+                }
+            )
+
+    def _create_elements(self, ids, texts, metadatas, embeddings):
```

Comment:
this should also do kwargs for custom tensors, also add maybe types?

Reply:
Will add types, but as I mentioned above I think this functionality can wait till next release

Reply:
kk